### DISCUSSION TITLE
App\Enums\TopicEnum::from(): Argument #1 ($value) must be of type int, string given

### DISCUSSION THREAD

**Discussion by artworkad** [livewire/livewire#6801] [7. Bugs (archived)]
### Livewire version

3.0.3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I use this rules, which worked for me in livewire 2:

```
public function rules(): array
{
    return [
        'note.topics' => 'array|min:1',
        'note.topics.*' => new Enum(TopicEnum::class),
        'note.date' => 'required|date_format:Y-m-d',
        'note.text' => 'required',
    ];
}
```
After upgrading I am getting the error

    App\Enums\TopicEnum::from(): Argument #1 ($value) must be of type int, string given

in 

    vendor / livewire / livewire / src / Mechanisms / HandleComponents / Synthesizers / EnumSynth .php : 34

This error pops up only when I remove a topic from the list of topics.

The body is:

```
{
    "_token": "avu27UX1G0fNux3D1QGpH2XPAU1ofsdlMF6J3QfL",
    "components": [
        {
            "snapshot": "{...},\"checksum\":\"...\"}",
            "updates": {
                "note.topics.1": "__rm__"
            },
            "calls": []
        }
    ]
}
```

Look like it is cause by the __rm__.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by johanmolen** [comment]
This issue also affects Filament.
